### PR TITLE
add walletClientName as prefix to all Gaia files mandatorily.

### PR DESCRIPTION
**Pull request by nandubatchu**

### CODE REVIEW

File: src/packages/gaia-service/index.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class GaiaService {
         const tokenFile = [blockstack.wrapProfileToken(token)];
         log.debug(tokenFile);
         try {
-            const finalUrl = await blockstack.uploadToGaiaHub(filename, JSON.stringify(tokenFile), hubConfig, "application/json");
+            const finalUrl = await blockstack.uploadToGaiaHub(`${prefix}_` + filename, JSON.stringify(tokenFile), hubConfig, "application/json");
```
Comment (SanchayPahalwani) [inline src/packages/gaia-service/index.ts]: As this is not being used we should not upload profile.json and remove this

File: src/packages/name-service/blockstack-service.ts [hunk 2]
```diff
@@ -186,7 +186,7 @@ export class BlockstackService extends nameService.NameService {
             throw ErrorHelper.getPackageError(PackageErrorCode.CouldNotFindKeyPairToRegisterName);
         }
 
-        await this._gaiaService.uploadProfileInfo(identityKeyPair.privKey);
+        await this._gaiaService.uploadProfileInfo(identityKeyPair.privKey, IdTranslator.blockstackDomainToCruxDomain(this._domain));
```
Comment (SanchayPahalwani) [inline src/packages/name-service/blockstack-service.ts]: this._domain should be renamed to this._blockstackdomain

Comment (nandubatchu) [reply src/packages/name-service/blockstack-service.ts]: I will be addressing this change in the upcoming PR.

File: src/packages/gaia-service/index.ts [hunk 3]
```diff
@@ -15,15 +15,15 @@ export class GaiaService {
...
+    public uploadContentToGaiaHub = async (filename: UPLOADABLE_JSON_FILES, privKey: string, content: any, prefix: string, type = "application/json"): Promise<string> => {
         const sanitizedPrivKey = sanitizePrivKey(privKey);
         const hubURL = this.gaiaWriteUrl;
         const hubConfig = await blockstack.connectToGaiaHub(hubURL, sanitizedPrivKey);
         const tokenFile = this._generateTokenFileForContent(sanitizedPrivKey, content);
         const contentToUpload: string = JSON.stringify(tokenFile);
         let finalURL: string;
         try {
-            finalURL = await blockstack.uploadToGaiaHub((prefix ? `${prefix}_` : "") + filename, contentToUpload, hubConfig, type);
+            finalURL = await blockstack.uploadToGaiaHub(`${prefix}_` + filename, contentToUpload, hubConfig, type);
```
Comment (quixote911) [inline src/packages/gaia-service/index.ts]: As discussed, lets move the prefix logic to BlockstackService class, and keep GaiaService as generic as possible.

File: src/packages/identity-utils.ts [hunk 4]
```diff
@@ -97,12 +97,18 @@ export class BlockstackId {
 }
 
 export class IdTranslator {
+    public static cruxDomainToBlockstackDomain = (domain: string): string => {
```
[the remark this answers is not in the document]
Comment (nandubatchu) [reply src/packages/identity-utils.ts]: this should be required in some scenarios to be a public method as discussed.

**Pull request merged**